File: pkg/pwrgrep/testdata/fixtures/javascript/external-input.ts
import fs from "fs";

function f(req, res) {
  // ruleid: js-external-input
  const home = process.env.HOME;
  // ruleid: js-external-input
  const args = process.argv;
  // ruleid: js-external-input
  const config = fs.readFileSync("config.json");
  // ruleid: js-external-input
  const id = req.params.id;
  // ruleid: js-external-input
  const q = location.search;
  // ok: js-external-input
  const local = "constant";
}

File: pkg/pwrgrep/testdata/fixtures/go/external-input.go
package fixture

import (
	"flag"
	"net/http"
	"os"
)

func f(w http.ResponseWriter, r *http.Request) {
	// ruleid: go-external-input
	home := os.Getenv("HOME")
	// ruleid: go-external-input
	first := os.Args[1]
	// ruleid: go-external-input
	port := flag.Int("port", 80, "")
	// ruleid: go-external-input
	body, _ := os.ReadFile("config.json")
	// ruleid: go-external-input
	name := r.FormValue("name")
	// ruleid: go-external-input
	q := r.URL.Query()
	// ok: go-external-input
	local := "constant"
	_, _, _, _, _, _, _ = home, first, port, body, name, q, local
}
